extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/StaticResourcesRecorder.java

```diff
@@ -0,0 +1,91 @@
+package io.quarkus.vertx.http.runtime;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+import io.quarkus.runtime.annotations.Recorder;
+import io.vertx.core.Handler;
+import io.vertx.ext.web.Route;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.StaticHandler;
+
+@Recorder
+public class StaticResourcesRecorder {
+
+    public static final String META_INF_RESOURCES = "META-INF/resources";
+
+    private static volatile Set<String> knownPaths;
+    private static volatile List<Path> hotDeploymentResourcePaths;
+
+    public static void setHotDeploymentResources(List<Path> resources) {
+        hotDeploymentResourcePaths = resources;
+    }
+
+    public void staticInit(Set<String> knownPaths) {
+        StaticResourcesRecorder.knownPaths = knownPaths;
+    }
+
+    public Consumer<Route> start() {
+
+        List<Handler<RoutingContext>> handlers = new ArrayList<>();
+
+        if (hotDeploymentResourcePaths != null && !hotDeploymentResourcePaths.isEmpty()) {
+            for (Path resourcePath : hotDeploymentResourcePaths) {
+                String root = resourcePath.toAbsolutePath().toString();
+                ThreadLocalHandler staticHandler = new ThreadLocalHandler(new Supplier<Handler<RoutingContext>>() {
```

Comment:
@stuartwdouglas @cescoffier I'm not quite sure the `ThreadLocalHandler` wrapper is still needed. It seems that the original thread-safety issue was fixed in vertx. The problem is that I have no idea how to test it properly...

Reply:
I can't remember why this was done. @stuartwdouglas any idea?

Reply:
If the underlying issue was fixed in Vert.x then it should be fine to remove.

Reply:
I believe that it was this one: https://github.com/vert-x3/vertx-web/issues/1429, fixed in vertx 3.8.4.

Reply:
I will remove the wrapper and let the CI run...